Find a functionally equivalent code:android.view.ViewGroup.dispatchHoverEvent:COMMENT
Method Modifier: protected   
Comment:/**
 * {@inheritDoc}
 */

@@ -15,26 +15,27 @@
         final float y = event.getY();
         final int childrenCount = mChildrenCount;
         if (childrenCount != 0) {
-            final boolean customChildOrder = isChildrenDrawingOrderEnabled();
+            final ArrayList<View> preorderedList = buildOrderedChildList();
+            final boolean customOrder = preorderedList == null && isChildrenDrawingOrderEnabled();
             final View[] children = mChildren;
             HoverTarget lastHoverTarget = null;
             for (int i = childrenCount - 1; i >= 0; i--) {
-                final int childIndex = customChildOrder ? getChildDrawingOrder(childrenCount, i) : i;
-                final View child = children[childIndex];
+                int childIndex = customOrder ? getChildDrawingOrder(childrenCount, i) : i;
+                final View child = (preorderedList == null) ? children[childIndex] : preorderedList.get(childIndex);
                 if (!canViewReceivePointerEvents(child) || !isTransformedTouchPointInView(x, y, child, null)) {
                     continue;
                 }
@@ -80,85 +81,88 @@
                     break;
                 }
             }
+            if (preorderedList != null)
+                preorderedList.clear();
         }
     }
     // Send exit events to all previously hovered children that are no longer hovered.

